src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs

```diff
@@ -1027,14 +1027,15 @@ public IEnumerable<Url> GetPageUrls(int tabId)
 
         public PageSettings GetPageSettings(int pageId, PortalSettings requestPortalSettings = null)
         {
-            var tab = GetPageDetails(pageId);            
+            var tab = GetPageDetails(pageId);
 
-            if (!_contentVerifier.IsContentExistsForRequestedPortal(tab.PortalID, requestPortalSettings))
+            var portalSettings = requestPortalSettings ?? PortalController.Instance.GetCurrentPortalSettings();
```

Comment:
I think that with this class we've already moved to constructor-injecting dependencies